uamqp/message.py

```diff
@@ -1235,13 +1334,32 @@ def __init__(self, header=None):
             self.priority = header.priority
 
     def __str__(self):
-        return str({
-            'delivery_count': self.delivery_count,
-            'time_to_live': self.time_to_live,
-            'first_acquirer': self.first_acquirer,
-            'durable': self.durable,
-            'priority': self.priority
-        })
+        return str(
+            {
+                "delivery_count": self.delivery_count,
+                "time_to_live": self.time_to_live,
+                "first_acquirer": self.first_acquirer,
+                "durable": self.durable,
+                "priority": self.priority,
+            }
+        )
+
+    def __getstate__(self):
+        state = self._get_properties_dict()
+        return state
+
+    def __setstate__(self, state):
+        for key, val in state.items():
+            self.__setattr__(key, val)
+
+    def _get_properties_dict(self):
+        return {
+            "delivery_count": self.delivery_count,
+            "time_to_live": self.time_to_live,
+            "first_acquirer": self.first_acquirer,
+            "durable": self.durable,
+            "priority": self.priority,
+        }
```

Comment:
wondering do we really need those for MessageHeader.. all the ivars are python objects